WordPress/Classes/Services/PostService.m

```diff
@@ -259,11 +259,17 @@ - (void)uploadPost:(AbstractPost *)post
             Post *postInContext = (Post *)[self.managedObjectContext existingObjectWithID:postObjectID error:nil];
             if (postInContext) {
                 postInContext.remoteStatus = AbstractPostRemoteStatusFailed;
-                // If the post was not created on the server yet we convert the post to a local draft post with the current date.
-                if (!postInContext.hasRemote) {
+                // If the post was not created on the server yet we convert the post to a local
+                // draft post with the current date. However, if it's PUBLISHED or DRAFT, we will
```

Comment:
This is super nitpicky, but it took me three reads of this comments to really understand it. Could we maybe reword the following section
>However, if it's PUBLISHED or DRAFT, we will

into something that talks about user intent, and not the state of a property?

so something like:

>However, if the post was supposed to be published or was a draft (e..g. has a `PostStatusPublish` or `PostStatusDraft`)

or something along those lines?

A sentence somewhere indicating something like "This is required because we _currently_ only want to automatically retry uploads of posts that are blah blah blah" near the end would also be great!

Reply:
@jklausa  Thanks for pointing that out. I left the original first sentence there without realizing how the whole paragraph reads. It is quite confusing. 

I'm not sure if this is better, but please let me know what you think: 

```swift
// If the post was not created on the server yet we convert the post to a local draft 
// with the current date. This post upload will be automatically retried later as a draft. 
// 
// However, if the post was supposed to be published or draft, we will leave it as is. 
// This is intentional because we currently want to automatically retry posts that 
// are either published or drafts. In the future, we will automatically retry all statuses. 
//
// Automatic uploads happen in `PostCoordinator.resume()`.
```

Reply:
that's much better, thank you!

Reply:
Thank you! Done in 76cfa6c9bac57fc96fa855951dcc7857cd2aa588.